docker/docker_image_src.go

```diff
@@ -161,6 +164,26 @@ func newImageSourceAttempt(ctx context.Context, sys *types.SystemContext, logica
 		client.Close()
 		return nil, err
 	}
+
+	if h, err := sysregistriesv2.AdditionalLayerStoreAuthHelper(endpointSys); err == nil && h != "" {
+		acf := map[string]struct {
```

Comment:
If this becomes a long-term ABI, it needs to be publicly documented as such. And maybe defined as a public Go type, to allow importing into consumers — although that might be necessary short-term.